src/libguac/string.c

```diff
@@ -44,6 +45,22 @@
  */
 #define REMAINING(n, length) (((n) < (length)) ? 0 : ((n) - (length)))
 
+size_t guac_itoa(char* restrict dest, int integer) {
+
+    /* Make sure memory is cleared. */
+    guac_mem_free(dest);
```

Comment:
What if somebody wants to write an integer value into something that can't be freed, like a stack-allocated character buffer?

Reply:
Yeah, that's a fair point. I could remove the freeing, but then I'd also have to remove the automatic allocation, and make the assumption that the caller has already allocated the correct amount of memory, right? I was trying to roll as much of that into this "convenience" function as possible, but I'm open to other suggestions if that makes sense.

From what little I can gather, it isn't possible to detect if something is heap or stack allocated?

Reply:
I don't think it's unreasonable to expect the caller to have to allocate enough space - that's pretty normal. 

If you do want to have this allocate space itself, consider having the length of the string passed in as a pointer the way that `dest` currently is, and having it return the newly allocated string. That way there won't be any chance of somebody passing something un-freeable into the function.

Reply:
> On balance I think the first option is still better though.

By which you mean, just ditch the automatic allocation and assume that they've allocated the space for it to begin with?

Reply:
Yes. That would allow the greatest flexibility in how this function could be used.

Reply:
Sounds good - should be implemented as suggested.

Reply:
I'm still seeing the `free()` call - did a commit get lost?

Reply:
No, just didn't remove it...fixed via rebase.

Reply:
Looks like it's still allocating the memory? I think we'd decided to let the caller allocate the memory for it, yeah?

Reply:
Yeah, sorry, you're correct. Let me fix that up...

Reply:
Okay, maybe now I've done it correctly?! LoL :facepalm: